.github/workflows/prbot.yml

```diff
@@ -1,18 +1,155 @@
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - run: |
+          npm ci
+          npx lerna bootstrap
+      - run: npm run lint:check
+  generate-doc:
+    name: "Generate doc"
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version-file: '.nvmrc'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - uses: actions/download-artifact@v2
+      - run: tar -xzvf artifact/build.tar.gz
+      - run: |
+          npm ci
+          npx lerna bootstrap
+      - run: npm run doc:generate
+  unit-test:
+    name: "Run unit tests"
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version-file: '.nvmrc'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - uses: actions/download-artifact@v2
+      - run: tar -xzvf artifact/build.tar.gz
+      - run: |
+          npm ci
+          npx lerna bootstrap
+      - run: npx lerna run copy:staticresources --scope @coveo/quantic
+      - run: npm test
+  e2e-test:
+    name: "Run e2e tests"
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        containers: [1, 2, 3]
+    env:
+      NODE_OPTIONS: '--max_old_space_size=4096'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version-file: '.nvmrc'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - uses: actions/download-artifact@v2
+      - run: tar -xzvf artifact/build.tar.gz
+      - run: |
+          npm ci
+          npx lerna bootstrap
+      - uses: cypress-io/github-action@2113e5bc19c45979ba123df6e07256d2aaba9a33
+        name: Run Cypress
+        with:
+          working-directory: ./packages/atomic
+          parallel: true
+          record: true
+          group: 'Atomic test runs'
+          start: npm run start:prod
+          wait-on: 'http://localhost:3333'
+          install: true
+        env:
+          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: cypress-screenshots
+          path: packages/atomic/cypress/screenshots
```

Comment:
This file feels a little unwieldy to read through. Could it make sense to split up the steps and reference them in here? Or would that be more complicated?